### PR TITLE
display decoded token

### PR DESCRIPTION
**Pull request by zburke**

### CODE REVIEW

File: src/settings/Token.js [hunk 1]
```diff
@@ -7,26 +7,72 @@ import { stripesShape } from '@folio/stripes/core';
...
+  /**
+   * decode the original token
+   */
+  componentDidMount = () => {
+    this.setState((state) => ({
+      cooked: this.parseToken(state.raw),
+    }));
   }
 
-  onSave(data) {
```
Comment (MikeTaylor) [inline src/settings/Token.js:22]: Nice to see the explicit `bind` going away `en passant`. Sheesh, what idiots we used to be back when we wrote code like that. Good thing we do everything exactly right these days.

File: src/settings/Token.js [hunk 2]
```diff
@@ -7,26 +7,72 @@ import { stripesShape } from '@folio/stripes/core';
...
+      raw: props.stripes.store.getState().okapi.token,
+      cooked: {},
+      error: '',
+    };
+  }
+
+  /**
+   * decode the original token
+   */
+  componentDidMount = () => {
```
Comment (MikeTaylor) [inline src/settings/Token.js:32]: What goes wrong if we just do this right in the constructor?

File: src/settings/Token.js [hunk 3]
```diff
@@ -7,26 +7,72 @@ import { stripesShape } from '@folio/stripes/core';
...
     stripes.setToken(data.token);
   }
 
+  /**
+   * decode a JWT token and save it in state.cooked. if decoding fails,
+   * write an error to state.error.
+   *
+   * verbatim from https://stackoverflow.com/questions/38552003/how-to-decode-jwt-token-in-javascript-without-using-a-library
+   */
+  parseToken = (t) => {
```
Comment (MikeTaylor) [inline src/settings/Token.js:49]: I'm just going to trust this code. No peer-review for you!

File: src/settings/Token.js [hunk 4]
```diff
@@ -38,9 +84,23 @@ class Token extends React.Component {
                 id="token"
                 name="token"
                 label={<FormattedMessage id="ui-developer.setToken.authenticationToken" />}
+                onChange={this.handleTokenChange}
```
Comment (MikeTaylor) [inline src/settings/Token.js:87]: I am not 100% convinced it's right to update this as soon as the field changes but delay the actual update of the token until the Save button is pressed. Don't you think both those things should happen at the same time?

Comment (zburke) [reply src/settings/Token.js:87]: This allows you to inspect the token before you save it. So, no, I don't think this should be tied to clicking "Save".

Comment (MikeTaylor) [reply src/settings/Token.js:87]: OK.

File: src/settings/Token.js (hunk 2 above)
Comment (zburke) [reply src/settings/Token.js:32]: `parseToken` itself writes the decoded token (or an error) to state, and you can't call `setState` on a component until it's fully mounted, hence doing that work here in `cDM`. Surely there's a way I could refactor that a bit to return a value or throw an exception, or even better refactor the whole component to be functional, but this seemed the least amount of work possible, and given this is totally unsanctioned work**, that seemed best. 

**I don't even remember why but for some reason I was decoding tokens the other day, I think related to the refresh-token work in FOLIO-1233, and it annoyed me that I had to go to a third-party site to look at my own data.

Comment (MikeTaylor) [reply src/settings/Token.js:32]: Yep, that all makes sense. Thanks again for doing this. It's great to see it now on https://folio-snapshot.aws.indexdata.com/settings/developer/token

**Pull request merged**